tf2_ros/include/tf2_ros/buffer.h

```diff
@@ -69,11 +74,43 @@ class Buffer : public BufferInterface, public AsyncBufferInterface, public tf2::
    * \param clock A clock to use for time and sleeping
    * \param cache_time How long to keep a history of transforms
    * \param node If passed advertise the view_frames service that exposes debugging information from the buffer
+   * \param  qos If passed change the quality of service of the frames_server_ service
    */
-  TF2_ROS_PUBLIC Buffer(
+  template<typename NodeT = rclcpp::Node::SharedPtr>
+  Buffer(
     rclcpp::Clock::SharedPtr clock,
     tf2::Duration cache_time = tf2::Duration(tf2::BUFFER_CORE_DEFAULT_CACHE_TIME),
-    rclcpp::Node::SharedPtr node = rclcpp::Node::SharedPtr());
+    NodeT && node = rclcpp::Node::SharedPtr(),
```

Comment:
Should this be:

```suggestion
    NodeT && node = NodeT(),
```

?